core/chainlink-nonroot.Dockerfile

```diff
@@ -0,0 +1,80 @@
+# MAKE ALL CHANGES WITHIN THE DEFAULT WORKDIR FOR YARN AND GO DEP CACHE HITS
+
+# THIS LINE IS AUTOGENERATED BY smartcontractkit/docker-cache/src/utils/update-docker-files.ts, DO NOT CHANGE MANUALLY
+FROM smartcontract/builder:1.0.38
+WORKDIR /chainlink
+# Have to reintroduce ENV vars from builder image
+ENV PATH /go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+
+COPY GNUmakefile VERSION ./
+COPY tools/bin/ldflags tools/bin/ldflags
+ARG COMMIT_SHA
+
+# Install yarn dependencies
+COPY yarn.lock package.json .yarnrc ./
+COPY patches patches
+COPY solc_bin solc_bin
+COPY .yarn .yarn
+COPY operator_ui/package.json ./operator_ui/
+COPY belt/package.json ./belt/
+COPY belt/bin ./belt/bin
+COPY evm-test-helpers/package.json ./evm-test-helpers/
+COPY evm-contracts/package.json ./evm-contracts/
+COPY tools/bin/restore-solc-cache ./tools/bin/restore-solc-cache
+RUN make yarndep
+
+
+COPY tsconfig.cjs.json tsconfig.es6.json ./
+COPY operator_ui ./operator_ui
+COPY belt ./belt
+COPY belt/bin ./belt/bin
+COPY evm-test-helpers ./evm-test-helpers
+COPY evm-contracts ./evm-contracts
+
+# Build operator-ui and the smart contracts
+RUN make contracts-operator-ui-build
+
+# Build the golang binary
+
+# THIS LINE IS AUTOGENERATED BY smartcontractkit/docker-cache/src/utils/update-docker-files.ts, DO NOT CHANGE MANUALLY
+FROM smartcontract/builder:1.0.38
+
+WORKDIR /chainlink
+
+# Have to reintroduce ENV vars from builder image
+ENV PATH /go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+
+COPY GNUmakefile VERSION ./
+COPY tools/bin/ldflags ./tools/bin/
+
+# Env vars needed for chainlink build
+ADD go.mod go.sum ./
+RUN go mod download
+
+# Env vars needed for chainlink build
+ARG COMMIT_SHA
+ARG ENVIRONMENT
+
+COPY --from=0 /chainlink/evm-contracts/abi ./evm-contracts/abi
+COPY --from=0 /chainlink/operator_ui/dist ./operator_ui/dist
+COPY core core
+COPY packr packr
+
+RUN make chainlink-build
+
+# Final layer: ubuntu with chainlink binary
+FROM ubuntu:18.04
+
+ENV DEBIAN_FRONTEND noninteractive
+RUN apt-get update && apt-get install -y ca-certificates
+
+COPY --from=1 /go/bin/chainlink /usr/local/bin/
+
+EXPOSE 6688
+
+RUN useradd --uid 14933 --create-home chainlink
+USER chainlink
```

Comment:
Are these the only two differences between root and non-root? If so, we should try to inject these as build-args instead, and continue maintaining one file.

Reply:
@HenryNguyen5 I think the plan is to deprecate the old root image after node operators switch over. Might be easier to just keep two files and then delete the old one. Thoughts? I don't feel strongly either way.

Reply:
Yeah, its like a 3 month period though, i'd imagine we'd change the dockerfile more often than that. We used to have another file for SGX that took awhile to get removed and was a pain to maintain

Reply:
sg man! I'll add the build args.